modules/shapes/assets/js/frontend/handlers/text-path.js

```diff
@@ -0,0 +1,217 @@
+				this.setOffset( startPoint.size );
+				this.setText( text );
+				break;
+
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * Attach a unique id to the path.
+	 *
+	 * @returns {void}
+	 */
+	attachIdToPath() {
+		const path = this.elements.svg.querySelector( '[data-path-anchor]' ) || this.elements.svg.querySelector( 'path' );
+		path.id = this.pathId;
+	}
+
+	/**
+	 * Initialize the text path element.
+	 *
+	 * @returns {void}
+	 */
+	initTextPath() {
+		const {
+			text,
+			startOffset,
+		} = this.elements.pathContainer.dataset;
+
+		this.attachIdToPath();
+
+		// Generate the `textPath` element with its settings.
+		this.elements.svg.innerHTML += `
+			<text>
+				<textPath id="${ this.textPathId }" href="#${ this.pathId }"></textPath>
+			</text>
+		`;
+
+		// Regenerate the elements object to have access to `this.elements.textPath`.
+		this.elements.textPath = this.elements.svg.querySelector( `#${ this.textPathId }` );
+
+		this.setOffset( startOffset );
+		this.setText( text );
+	}
+
+	/**
+	 * Set the new text into the path.
+	 *
+	 * @param newText {string} The new text to put in the text path.
+	 *
+	 * @returns {void}
+	 */
+	setText( newText ) {
+		let {
+			href,
+			target,
+			rel,
+		} = this.elements.pathContainer.dataset;
+
+		// Add link attributes.
+		if ( href ) {
+			newText = `<a href="${ href }" rel="${ rel }" target="${ target }">${ newText }</a>`;
+		}
+
+		// Set the text.
+		this.elements.textPath.innerHTML = newText;
+
+		// Remove the cloned element if exists.
+		const existingClone = this.elements.svg.querySelector( `#${ this.textPathId }-clone` );
+
+		if ( existingClone ) {
+			existingClone.remove();
+		}
+
+		// Reverse the text if needed.
+		if ( this.shouldReverseText() ) {
+			// Keep an invisible selectable copy of original element for better a11y.
+			const clone = this.elements.textPath.cloneNode();
+			clone.id += '-clone';
+			clone.classList.add( 'elementor-hidden' );
+			clone.textContent = newText;
+
+			this.elements.textPath.parentNode.appendChild( clone );
+			this.reverseToRTL();
+		}
+	}
+
+	/**
+	 * Determine if the current layout should be RTL.
+	 *
+	 * @returns {boolean}
+	 */
+	isRTL() {
+		const { text_path_direction: direction } = this.getElementSettings();
+		let isRTL = elementorCommon.config.isRTL;
+
+		if ( direction ) {
+			isRTL = ( 'rtl' === direction );
+		}
+
+		return isRTL;
+	}
+
+	/**
+	 * Determine if it should RTL the text (reversing it, etc.).
+	 *
+	 * @returns {boolean}
+	 */
+	shouldReverseText() {
+		return ( this.isRTL() && -1 === navigator.userAgent.indexOf( 'Firefox' ) );
+	}
+
+	/**
+	 * Reverse the text path to support RTL.
+	 *
+	 * @returns {void}
+	 */
+	reverseToRTL() {
+		// Make sure to use the inner `a` tag if exists.
+		let parentElement = this.elements.textPath;
+		parentElement = parentElement.querySelector( 'a' ) || parentElement;
+
+		const pattern = /([\u0591-\u07FF\u200F\u202B\u202E\uFB1D-\uFDFD\uFE70-\uFEFC\s$&+,:;=?@#|'<>.^*()%!-]+)/ig;
```

Comment:
Explain. + try mix en+he+en.

Reply:
Mixing he + en in the control value? Already tried that. It works with some small bugs. 
This behavior is approved by the Product.